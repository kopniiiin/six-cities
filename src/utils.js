const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`
];

export const extend = (target, update) => Object.assign({}, target, update);

export const pushElement = (array, element) => [...array, element];

export const replaceElement = (array, element, index) => [...array.slice(0, index), element, ...array.slice(index + 1)];

export const removeElement = (array, index) => [...array.slice(0, index), ...array.slice(index + 1)];

export const upperCaseFirstLetter = (string) => string[0].toUpperCase() + string.slice(1);

export const formatDate = (dateString) => {
  const date = new Date(dateString);
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
