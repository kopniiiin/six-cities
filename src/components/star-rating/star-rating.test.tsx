import * as React from "react";
import {configure, shallow} from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import {extend} from "../../utils";

import StarRating from "./star-rating";

configure({adapter: new Adapter()});

const testMocks = {
  blockClassName: `block`,
  value: 4,
  isValueShown: false
};

describe(`snapshot test: StarRating component`, () => {
  it(`should render correctly`, () => expect(
      toJson(shallow(<StarRating {...testMocks}/>))
  ).toMatchSnapshot());

  it(`should render value`, () => expect(
      toJson(shallow(<StarRating {...extend(testMocks, {isValueShown: true})}/>))
  ).toMatchSnapshot());
});
