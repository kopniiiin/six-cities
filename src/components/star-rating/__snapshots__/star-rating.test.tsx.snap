// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: StarRating component should render correctly 1`] = `
<div
  className="block__rating rating"
>
  <div
    className="block__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;

exports[`snapshot test: StarRating component should render value 1`] = `
<div
  className="block__rating rating"
>
  <div
    className="block__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
  <span
    className="block__rating-value rating__value"
  >
    4
  </span>
</div>
`;
